index.js

```diff
@@ -38,33 +40,62 @@ const button = ActionButton({ // eslint-disable-line new-cap
   }
 });
 
+function setupApp() {
+  app = new Router(mod);
 
-function sendToWeb(data) {
-  mod.port.emit('from-addon-to-web', data);
-}
+  if (self.loadReason === 'install') {
+    app.send('addon-self:installed');
+  } else if (self.loadReason === 'enable') {
+    app.send('addon-self:enabled');
+  } else if (self.loadReason === 'upgrade') {
+    app.send('addon-self:upgraded');
+  }
```

Comment:
These are the only reasons we really care about https://developer.mozilla.org/en-US/Add-ons/SDK/High-Level_APIs/self#loadReason. I don't think there would be any other reason to log these.